lib/signs_ui/config/expiration.ex

```diff
@@ -59,6 +59,59 @@ defmodule SignsUi.Config.Expiration do
     |> Enum.into(%{})
   end
 
+  @spec expire_signs_via_alert(list(SignsUi.Config.State.t()), MapSet.t()) ::
+          list(SignsUi.Config.State.t())
+  def expire_signs_via_alert(sign_states, alert_ids) do
+    expired_sign_states =
+      get_expired_sign_states(
+        sign_states,
+        alert_ids,
+        []
+      )
+
+    expired_sign_states
+  end
+
+  @spec get_expired_sign_states(
+          list(SignsUi.Config.State.t()),
+          MapSet.t(),
+          list(SignsUi.Config.State.t())
+        ) :: list(SignsUi.Config.State.t())
+  def get_expired_sign_states(remaining_sign_states, alert_ids, expired_sign_states) do
+    if remaining_sign_states == [] do
+      expired_sign_states
+    else
+      [sign_state | tail] = remaining_sign_states
+
+      cond do
+        sign_state.config.mode == :auto ->
+          get_expired_sign_states(tail, alert_ids, expired_sign_states)
+
+        Enum.member?(alert_ids, sign_state.config.alert_id) ->
+          get_expired_sign_states(tail, alert_ids, expired_sign_states)
+
+        true ->
+          new_sign_state = %SignsUi.Config.Sign{
+            id: sign_state.config.alert_id,
+            config: %{mode: :auto}
+          }
```

Comment:
Is there a reason to create a new `Sign` vs updating the existing one?
```elixir
sign_state = %{sign_state | config:  %{mode: :auto}}
```

Reply:
There's also an existing `expire_single_sign` function you could use.

Reply:
I'm creating a new sign because that's what @losvedir asked for. "`expire_signs_via_alerts([signs...], alert_ids)` should return a list of the signs that have an `alert_id` which is not present in `alert_ids`. and those signs should be new signs with the same ID, but in auto mode"